### DISCUSSION TITLE
why is my params not optimizring

### DISCUSSION THREAD

**Discussion by krshailendra27121983-jpg** [mrdbourke/pytorch-deep-learning#1319] [Q&A] [answered]
so i was watching the PyTorch for Deep Learning & Machine Learning – Full Course (https://www.youtube.com/watch?v=V_xro1bcAuA)
and once i got to optimizering my model params it did not work ( not i will spell a lot of stuff wrong because i am 12 years old)
but i was at 6:46:39 in the video the weights and biases were not optimizeing like they did in his video so can anyone fix my ( admittly shit ) code 

torch.manual_seed(42)
epochs = 1

for i in range(epochs):
  model_0.train()
  y_pred = model_0(X_train)
  loss = loss_fn(y_pred, y_train)

  optim.zero_grad()
  loss.backward()
  optim.step()

  model_0.eval()
model_0.state_dict()
OrderedDict([('weights', tensor([0.3406])), ('bias', tensor([0.1388]))]) (?)

**Comment by krshailendra27121983-jpg** [comment, marked as answer]
never mind littery 5 minutes after i made this comment i found out i just was running the full code insted of the single cell i sould have been running so yay........................... :)

**Reply by mrdbourke** [reply]
oh dam nice find! We've all been there!

I often find that writing out a problem helps.

And so cool you're getting into deep learning at 12! Very inspiring to see.
